ui/apps/platform/src/Containers/ComplianceEnhanced/Coverage/ClusterDetailsPage.tsx

```diff
@@ -1,26 +1,113 @@
+            <PageSection variant="light" className="pf-v5-u-py-md">
+                <Breadcrumb>
+                    <BreadcrumbItem>Compliance coverage</BreadcrumbItem>
+                    <BreadcrumbItemLink
+                        to={generatePath(coverageProfileClustersPath, {
+                            profileName,
+                        })}
+                    >
+                        Clusters
+                    </BreadcrumbItemLink>
+                    <BreadcrumbItem isActive>
+                        {isLoadingClusterProfileData ? (
+                            <Skeleton screenreaderText="Loading cluster name" width="150px" />
+                        ) : (
+                            // TODO: placeholder until we get cluster name
+                            clusterId
+                        )}
+                    </BreadcrumbItem>
+                </Breadcrumb>
+            </PageSection>
+            <Divider component="div" />
             <PageSection variant="light">
-                <div>ClusterDetailsPage</div>
-                <Button
-                    onClick={() => {
-                        history.push(
-                            `${complianceEnhancedCoveragePath}/profiles/${profileName}/clusters`
-                        );
-                    }}
-                    variant="primary"
+                <Flex
+                    direction={{ default: 'column' }}
+                    alignItems={{ default: 'alignItemsFlexStart' }}
                 >
-                    Go to all clusters (ProfileClustersPage)
-                </Button>
+                    <Title headingLevel="h1" className="pf-v5-u-w-100">
+                        {isLoadingClusterProfileData ? (
+                            <Skeleton fontSize="2xl" screenreaderText="Loading cluster name" />
+                        ) : (
+                            // TODO: placeholder until we get cluster name
+                            clusterId
+                        )}
+                    </Title>
+                    <LabelGroup numLabels={1}>
+                        <Label>
+                            {isLoadingClusterProfileData ? (
+                                <Skeleton
+                                    screenreaderText="Loading number of profiles scanned on cluster"
+                                    width="135px"
+                                />
+                            ) : (
+                                `Scanned by: ${clusterProfileData?.totalCount} profiles`
+                            )}
+                        </Label>
+                    </LabelGroup>
+                </Flex>
+            </PageSection>
+            <Divider component="div" />
+            <PageSection>
+                <ProfilesToggleGroup
+                    profiles={clusterProfileData?.scanStats || []}
+                    route={coverageClusterDetailsPath.replace(':clusterId', clusterId)}
```

Comment:
You might be able to call `safeGeneratePath` here and elsewhere, after it has been merged:
https://github.com/stackrox/stackrox/pull/11350/files#diff-0a19d066a4504eb4d6b668e234f03f400da7485b832d401fca70fd6f538f9602R40-R55